### PR TITLE
Fix #2753

### LINKED ISSUE #2753: Fulfillment of Armor - Shouldn't take your armor
When a guard request armor and you click on fulfill.  It will fulfill that request with the armor you are wearing.

Armor your wearing shouldn't be possible to fulfill the request for item.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/61766033-fulfillment-of-armor-shouldn-t-take-your-armor?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ghost**
Closes #2753 

Prevents guards from taking equipped armor during fufillment

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 1]
```diff
@@ -580,6 +580,11 @@ private void fulfill(@NotNull final Button button)
             {
                 itemStack = inventory.getStackInSlot(InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(inventory), requestPredicate));
             }
+
+            for (int checkInv = 0; checkInv <= 3; checkInv++)
+                if (itemStack.isItemEqual(inventory.armorInventory.get(checkInv))) {
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: But like this, I think it won't work if the player has the armor in his armor slots + the armor in his inventory.
+ Can you put brackets around the for and the brackets for the if on an extra line?

Comment (ghost) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: In my tests it seems to work with armor equipped and in the inventory

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — APPROVED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Pull request closed without merging**